### PR TITLE
[PPML]add gramine bigdata toolkit

### PR DESCRIPTION
**Pull request by ShanSimu**
## Description

Create a gramine image only with bigdata frameworks, including hadoop, spark and hive.

##  How to test?
As described in README.md

### CODE REVIEW

File: ppml/trusted-bigdata/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+    mkdir -p /ppml/work/password && \
+    mkdir -p /ppml/work/data && \
+    mkdir -p /ppml/work/models && \
+    mkdir -p /ppml/work/apps && \
+    mkdir -p /root/.keras/datasets && \
+    mkdir -p /root/.zinc && \
+    mkdir -p /root/.m2 && \
+    mkdir -p /root/.kube/
+
+COPY --from=bigdata /opt/jdk  /opt/jdk8
```
Comment (glorysdj) [inline ppml/trusted-bigdata/Dockerfile:198]: jdk is not needed after xiangyu's pr

File: ppml/trusted-bigdata/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+COPY --from=bigdata /opt/scala-${SCALA_VERSION}.${SCALA_SUB_VERSION_NUM}  /opt/scala-${SCALA_VERSION}.${SCALA_SUB_VERSION_NUM}
```
Comment (glorysdj) [inline ppml/trusted-bigdata/Dockerfile:199]: do we need scala?

File: ppml/trusted-bigdata/settings.xml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+    <settings xmlns="http://maven.apache.org/SETTINGS/1.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (glorysdj) [inline ppml/trusted-bigdata/settings.xml:1]: why need maven settings? this can not be committed

File: ppml/trusted-bigdata/Dockerfile (hunk 1 above)
Comment (ShanSimu) [reply ppml/trusted-bigdata/Dockerfile:198]: jdk will be delete after xiangyu'pr merge

File: ppml/trusted-bigdata/settings.xml (hunk 3 above)
Comment (ShanSimu) [reply ppml/trusted-bigdata/settings.xml:1]: This is to speed up mvn, it will be deleted before merge

**Comment by glorysdj** [comment]
we should remove all flink, since flink also need a entrypoint, we can not put spark and flink together in a image

**Comment by glorysdj** [comment]
>

**Comment by glorysdj** [comment]
> Do we want to release separate packages for different Big Data framework? Will it be too many if we want to support more frameworks in future?

There will be 5 images for bigdl-ppml, bigdata, and deep learning as listed below. Is this OK?
- bigdl-ppml-gramine-base
  - bigdl-ppml (legacy bigdl-ppml-trusted-big-data-ml-python, +FL) 
  - bigdl-ppml-trusted-deep-learning (pytorch/tf etc.)
  - bigdl-ppml-trusted-bigdata (spark on k8s-sgx, hdfs, hive)
  - bigdl-ppml-trusted-realtime (flink on k8s-sgx, kafka)
  - bigdl-ppml-trusted-python (flask, torchserv, etc)

File: ppml/trusted-bigdata/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+# stage.2 gramine
+FROM $BASE_IMAGE_NAME:$BASE_IMAGE_TAG
+
+ARG SPARK_VERSION
+ARG TINI_VERSION
+
+ENV SPARK_VERSION                       ${SPARK_VERSION}
+ENV SPARK_HOME                          /ppml/spark-${SPARK_VERSION}
+ENV LOCAL_IP                            127.0.0.1
+ENV SPARK_MASTER_IP                     127.0.0.1
```
Comment (glorysdj) [inline ppml/trusted-bigdata/Dockerfile:129]: from SPARK_MASTER_IP to SPARK_BLOCK_MANAGER_PORT, we can delete them

File: ppml/trusted-bigdata/spark-driver-template.yaml [hunk 5]
```diff
@@ -0,0 +1,62 @@
+apiVersion: v1
+kind: Pod
+spec:
+  containers:
+  - name: spark-driver
+    securityContext:
+      privileged: true
```
Comment (glorysdj) [inline ppml/trusted-bigdata/spark-driver-template.yaml:7]: please try non-privileged mode

File: ppml/trusted-bigdata/Dockerfile (hunk 4 above)
Comment (ShanSimu) [reply ppml/trusted-bigdata/Dockerfile]: deleted

File: ppml/trusted-bigdata/spark-driver-template.yaml (hunk 5 above)
Comment (ShanSimu) [reply ppml/trusted-bigdata/spark-driver-template.yaml]: done

File: ppml/trusted-bigdata/spark-executor-template.yaml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+      - name: nfs-storage
+        mountPath: /ppml/data
+      - name: nfs-storage
+        mountPath: /root/.kube/config
+        subPath: kubeconfig
+    resources:
+      requests:
+        cpu: 16
+        memory: 128Gi
+        sgx.intel.com/epc: 133258905600
```
Comment (glorysdj) [inline ppml/trusted-bigdata/spark-executor-template.yaml:37]: 133258905600 is too big, can we set it as the same size of the epc mem?

Comment (ShanSimu) [reply ppml/trusted-bigdata/spark-executor-template.yaml]: fixed

**Review by glorysdj — APPROVED** [review]
LGTM

**Pull request merged**